### PR TITLE
CICD: docs github actions

### PR DESCRIPTION
**Pull request by maffettone**
Add publishing to docs GHA. Also enable manual trigger with workflow dispatch. 

## Description

Mostly borrowed from https://github.com/bluesky/ophyd/blob/master/.github/workflows/docs.yml. 

## Motivation and Context

Docs were previously unpublished, and people have been left unaware of developments.

### CODE REVIEW

**Comment by maffettone** [comment]
@mrakitin Looking for a sanity check on this, because I think you led the changes for the ophyd docs and cookie-cutter.  (The test failures are expected, working on those in parallel).

File: .github/workflows/docs.yml
Comment (mrakitin) [inline .github/workflows/docs.yml:1]: I'd bump the tags for those 2 actions to v4 for both (coincidence!):
```yaml
...
- uses: actions/checkout@v2
...
- uses: actions/setup-python@v2
```

**Review by mrakitin — APPROVED** [review]
Looks good! Left a minor suggestion to keep up with the versions of the actions.

File: .github/workflows/docs.yml
Comment (maffettone) [reply .github/workflows/docs.yml:1]: Made these adjustments. It should be ready to merge after CI

**Pull request merged**